src/components/TreeView.js

```diff
@@ -34,8 +48,34 @@ export default class TreeView extends React.Component {
   }
 }
 
+function checkSignatures(nodes, fetchedSigners) {
+  // catch fetch signature errors
+  if (fetchedSigners === null) {
+    nodes[1].state = SIGNATURE.NOT_VERIFIED_YET;
```

Comment:
I think something is wrong here. I set connection latency to 10s in devtools. I expected to see "Checking signatures..." but instead I got "Error checking signatures..." and after 10s it disappeared and updated each signature state.

Reply:
Ah I see. The error is that I have an initial state of null with only two states, fail and success. However, there is in fact also a third state (different from initial state) for loading.